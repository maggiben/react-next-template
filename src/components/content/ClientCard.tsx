import styled from "styled-components";
import { 
  UserIcon, 
  CheckCircleIcon, 
  TableView, 
  Column, 
  Cell, 
  CloseCircleIcon, 
  QuestionCircleIcon, 
  Grid, 
  GridItem, 
  Heading, 
  Label,
} from '@fravega-it/bumeran-ds-fvg'

import { Person } from './index';
import { Children } from "react";

const Card = styled.div`
  display: block;
  width: 100%;
  border-width: 1px;
  border-style: solid;
  border-color: ${({ theme }) => theme.colors.neutral[300]};
  border-radius: ${({ theme }) => theme.borderRadius.m};
`;

const Centered = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

type ClientCardProps = {
  isOpen?: boolean;
  person: Person;
}
  
const ClientCard = (props: ClientCardProps) => {
  const { person } = props;
  return (
    <Card>
      <Grid>
        <GridItem xs={6} justifySelf="start" alignSelf="center">
          <Centered>
            <UserIcon size="l" color="violet" colorTone="600" /><Heading size="s">Name: {person.name}</Heading>
          </Centered>
        </GridItem>
        <GridItem xs={6} alignSelf="center" justifySelf="center">
          <Centered>
            <QuestionCircleIcon size="l" color="violet" colorTone="600" /><Heading size="s">Estatus:</Heading><Label leftIcon={person.profession ? <CheckCircleIcon size="s" /> : <CloseCircleIcon size="s"/> } label={person.profession} color={person.profession ? "green" : "red"}/>
          </Centered>
        </GridItem>
        <GridItem xs={12}>
        <TableView
            items={[person]}
            renderColumns={() => (
              <>
                <Column minWidth={50} label="name" />
                <Column minWidth={100} label="faceapi" />
                <Column minWidth={70} label="email" />
                <Column minWidth={70} label="email confirmed" />
                <Column minWidth={100} label="city" />
              </>
            )}
            renderCells={({ id, name, faceapi, email, city }) => (
              <>
                <Cell>{name}</Cell>
                {/* <Cell><Button id={id} label={faceapi.toString()} /></Cell> */}
                <Cell>
                  {/* { (function () {
                      if(faceapi) {
                        return (
                          <input type="checkbox" id={id} checked={faceapi} readOnly/>
                        );
                      } else {
                        return <Button id={id} size="s" label="Resend" />
                      }
                    })()
                  } */}
                  <Label leftIcon={faceapi ? <CheckCircleIcon size="s" /> : <CloseCircleIcon size="s"/> } label={faceapi.toString()} color={faceapi ? "green" : "red"}/>
                  </Cell>
                {/* <Cell><Label label={faceapi.toString()} color={faceapi ? "green" : "red"}/></Cell> */}
                <Cell>{email.address}</Cell>
                {/* <Cell><input type="checkbox" id={id} checked={email.confirmed} readOnly/></Cell> */}
                <Cell>
                  <Label leftIcon={email.confirmed ? <CheckCircleIcon size="s" /> : <CloseCircleIcon size="s"/> } label={email.confirmed.toString()} color={email.confirmed ? "green" : "red"}/>
                </Cell>
                {/* <Cell><Label label={email.confirmed.toString()} color={email.confirmed ? "green" : "red"}/></Cell> */}
                <Cell>{city}</Cell>
              </>
            )}
          />
        </GridItem>
      </Grid>
    </Card>
  )
};

  export default ClientCard;
