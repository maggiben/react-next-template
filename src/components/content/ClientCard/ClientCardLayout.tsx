import styled from "styled-components";
import { 
  Grid, 
  GridItem, 
  Heading, 
  Label,
  Button,
} from '@fravega-it/bumeran-ds-fvg'
import { useTranslation } from 'react-i18next';
import { Person } from 'types/type';
import { useMediaQuery } from 'usehooks-ts';
import DataContainer from '@components/DataContainer/DataContainer';
import ValidationList from '../ValidationList/ValidationList';
import { SpaceRight, SpaceTop } from '@components/Spacing/Spacing';

const Card = styled.div`
  position: relative;
  display: block;
  width: 100%;
  border-width: 1px;
  border-style: solid;
  padding: ${({ theme }) => theme.spacing.m};
  background-color: ${({ theme }) => theme.colors.white};
  border-color: ${({ theme }) => theme.colors.neutral[300]};
  border-radius: ${({ theme }) => theme.borderRadius.m};
`;

const Centered = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

type ClientCardLayoutProps = {
  person: Person;
}

const ClientCardLayout = (props: ClientCardLayoutProps) => {
  const { t } = useTranslation();
  const { person } = props;
  const columns = useMediaQuery('(min-width: 768px)') ? 3 : 1;

  const clientData: Record<string, string>[] = [
    {[t('birth date')]: new Date(person.age).toLocaleDateString()}, 
    {[t('email')]: person.email.address}, 
    {[t('address')]: `${person.address.street} • ${person.address.number} • ${person.address.city}`}, 
    {[t('phone')]: `${person.phone.codeArea.toString()} - ${person.phone.number.toString()}`},
    {[t(person.identification.type)]: person.identification.number.toString()}, 
    {[t('id expiration date')]: new Date(person.identification.expiration).toLocaleDateString()}, 
  ];

  /* TODO: quick fix make batter */
  const colors = {
    onboardingfull: true,
    onboardingincompleto: false,
    dnicaduco: false,
  };

  const labels = {
    onboardingfull: t('onboarding full'),
    onboardingincompleto: t('onboarding incomplete'),
    dnicaduco: t('dni expired'),
  };

  return (
    <Card>
      <Grid>
        <GridItem xs={6} justifySelf="start" alignSelf="center">
          <Centered>
            <Heading size="s">{person.name} {person.lastname}</Heading><SpaceRight size="s" /><Label label={labels[person.status]} color={colors[person.status] ? 'green' : 'red'}/>
          </Centered>
        </GridItem>
        <GridItem xs={6} alignSelf="center" justifySelf="end">
          <Button label={t('resend email')} variant="primary" size="s" onClick={() => alert('hola')}/>
        </GridItem>
        <GridItem xs={12}>
          <ValidationList person={person} />
          <SpaceTop size="s" />
          <DataContainer data={clientData} columns={columns} />
        </GridItem>
      </Grid>
    </Card>
  )
};

  export default ClientCardLayout;
