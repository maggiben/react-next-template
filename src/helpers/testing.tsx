import { default as baseRenderer } from 'react-test-renderer';
import { render, RenderResult } from '@testing-library/react';
import i18n from '@utils/i18n';
import { I18nextProvider } from 'react-i18next';
import { RecoilRoot } from 'recoil';
import { ThemeProvider } from 'styled-components';
import { theme, IDefaultTheme } from '@fravega-it/bumeran-ds-fvg';
import React, { ReactElement } from 'react';
import '@testing-library/jest-dom'

/**
 * Helper used to render components for testing.
 * The function appends the theme to the children.
 * */
interface IRenderer {
  create: (children: ReactElement) => RenderResult;
  baseRenderer: (children: ReactElement) => baseRenderer.ReactTestRenderer;
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme) => RenderResult;
  renderWithI18n: (children: ReactElement) => RenderResult;
}

export const renderer: IRenderer = {
  create: (children: ReactElement): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  baseRenderer: (children: ReactElement): baseRenderer.ReactTestRenderer =>
    baseRenderer.create(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithI18n: (children: ReactElement): RenderResult => 
    render(
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
      </ThemeProvider>
  ),
};