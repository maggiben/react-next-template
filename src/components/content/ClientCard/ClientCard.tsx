import { useState, useEffect, useMemo, useCallback, useRef } from 'react';
import { useRecoilState, useRecoilValue } from 'recoil';
import DuplicateModal from '../DuplicateModal/DuplicateModal';
import Waiting from '@components/Waiting';
import { useRouter } from 'next/router';

import { Person } from 'types/type';
import { useTranslation } from 'react-i18next';
import { useFetch } from 'usehooks-ts';
import ClientCardLayout from './ClientCardLayout';
import NoResults from '@components/NoResults';
import { personState, personsState } from '@states/atoms';
import {
  hasSelectedPerson as hasSelectedPersonSelector,
  hasMultiplePerson as hasMultiplePersonSelector
} from '@states/selectors';
import * as array from '@utils/array';

const ClientCard = () => {
  const { t } = useTranslation();
  const router = useRouter();
  const [person, setPerson] = useRecoilState(personState);
  const [persons, setPersons] = useRecoilState(personsState);
  const hasSelectedPerson = useRecoilValue(hasSelectedPersonSelector);
  const hasMultiplePerson = useRecoilValue(hasMultiplePersonSelector);
  const [isOpenModal, setIsOpenModal] = useState(false);
  const searchParams = useMemo(() => {
    return new URLSearchParams(router.query as Record<string, string>);
  }, [router.query]);
  const { data, error } = useFetch<Person[]>(`api/searchx?${searchParams.toString()}`);

  const closeModal = useCallback((): void =>  {
    // setPersons(undefined);
    setIsOpenModal(false);
    // erase search history
    router.push({
      pathname: document.location.pathname,
    });
  }, []);

  const onSelectPersonModal = useCallback((data?: Person): void =>  {
    setIsOpenModal(false);
    if (data) {
      setPerson(data);
    }
  }, []);

  // useEffect(() => {
  //   if (error) {
  //     router.push({
  //       pathname: `/${error.cause ?? 404}`,
  //       query: {
  //         message: error.message,
  //         oldSearchParams: searchParams.toString(),
  //       },
  //     });
  //   }
  // }, [error]);
    
  useEffect(() => {
    /*
      Quitamos la prop: 'selected' que es dinamica (se usa en los radio buttons)
      El resto de los objetos deberia permanecer inmutado y la comparacion
      ser equivalente
     */

    const a = array.removeProperty(data ?? [], 'selected');
    const b = array.removeProperty(persons ?? [], 'selected');

    if (data && !array.isEqual(a, b)) {
      setPersons(data);
      if (data.length === 1) {
        setIsOpenModal(false);
        setPerson(data[0]);
      }
    }
  }, [ data ]);

  useEffect(() => {
    if (hasMultiplePerson && !hasSelectedPerson && !person ) {
      setIsOpenModal(true);
    } 
  }, [ hasMultiplePerson, hasSelectedPerson ]);

  const noResultsMessage = useMemo(() => searchParams.get('documentNumber') || searchParams.get('email') || error?.message, [error]);
  return (
    <div data-testid="client-card">
      { persons && isOpenModal && <DuplicateModal isOpen={isOpenModal} closeModal={closeModal} onSelectPersonModal={onSelectPersonModal}/> }
      { person && <ClientCardLayout person={person} /> }
      { !person && !persons && !error && <Waiting message={t('searching')} /> }
      { error && <NoResults message={noResultsMessage} /> }
    </div>
  )
};

  export default ClientCard;
