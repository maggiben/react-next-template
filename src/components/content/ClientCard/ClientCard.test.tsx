import React from 'react';
import { renderer } from '@helpers/testing';
import 'jest-styled-components';
import ClientCard from '.';

import { expect } from '@jest/globals';

test('Check ClientCard works', () => {
    const tree = renderer.create(<ClientCard />).toJSON();
    expect(tree).toMatchSnapshot();
});
