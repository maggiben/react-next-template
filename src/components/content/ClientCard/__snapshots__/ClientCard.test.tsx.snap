// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check ClientCard works 1`] = `
.c9 {
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  font-family: Roboto;
  color: inherit;
}

.c25 {
  all: unset;
  display: block;
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  font-family: Roboto;
  color: inherit;
}

.c5 {
  font-size: 16px;
  line-height: 24px;
  font-weight: 600;
  font-family: Work Sans;
  color: #242433;
}

.c18 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 600;
  font-family: Work Sans;
  color: inherit;
}

.c4 {
  fill: #371380;
}

.c8 {
  fill: #105225;
}

.c12 {
  fill: #8D8DA6;
}

.c11 {
  color: #65657E;
  background-color: #FFFFFF;
  border: 1px solid #D5D5E7;
  max-height: 36px;
  padding: calc(8px - 1px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  border-radius: 8px;
}

.c11 svg {
  fill: #65657E;
}

.c11:hover {
  border: 1px solid #8D8DA6;
  background-color: #FAFAFF;
  color: #3B3B4E;
}

.c11:hover svg {
  fill: #3B3B4E;
}

.c11:focus-visible {
  border: 1px solid #8D8DA6;
  background-color: #FAFAFF;
  color: #3B3B4E;
  box-shadow: 0px 0px 0px 4px #B3D1FF;
  outline: unset;
}

.c11:focus-visible svg {
  fill: #3B3B4E;
}

.c11:active {
  background-color: #F0F0FA;
  border: 1px solid #65657E;
  color: #242433;
}

.c11 label {
  cursor: pointer;
}

.c11:disabled,
.c11 [disabled] {
  cursor: unset;
  background-color: #F0F0FA;
  color: #D5D5E7;
  border: unset;
  border: solid 1px transparent;
}

.c11:disabled label,
.c11 [disabled] label {
  cursor: unset;
}

.c11:disabled:hover,
.c11 [disabled]:hover {
  border: solid 1px transparent;
}

.c11:disabled svg,
.c11 [disabled] svg {
  fill: #D5D5E7;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(12,1fr);
  grid-column-gap: 16px;
  grid-row-gap: 16px;
  max-width: 1224px;
  margin: auto;
}

.c2 {
  justify-self: start;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  grid-column: span 6;
}

.c6 {
  justify-self: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  grid-column: span 5;
}

.c10 {
  justify-self: end;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  grid-column: span 1;
}

.c13 {
  grid-column: span 12;
}

.c7 {
  padding: 2px 12px 2px 8px;
  background-color: #E0FFEA;
  color: #105225;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  border-radius: 1000px;
}

.c24 {
  padding: 8px 12px;
}

.c16 {
  font-weight: unset;
  min-width: 50px;
  text-align: unset;
  background-color: #F0F0FA;
  height: 56px;
}

.c19 {
  font-weight: unset;
  min-width: 100px;
  text-align: unset;
  background-color: #F0F0FA;
  height: 56px;
}

.c20 {
  font-weight: unset;
  min-width: 70px;
  text-align: unset;
  background-color: #F0F0FA;
  height: 56px;
}

.c21 {
  font-weight: unset;
  min-width: 40px;
  text-align: unset;
  background-color: #F0F0FA;
  height: 56px;
}

.c17 {
  padding: 8px 12px;
}

.c14 {
  width: 100%;
  border-collapse: separate;
  border-spacing: 0;
}

.c14 tr th:last-child,
.c14 tr td:last-child {
  border-right: 0.5px solid #D5D5E7;
}

.c14 tr td,
.c14 tr th {
  border-bottom: 0.5px solid #D5D5E7;
}

.c14 tr th:first-child,
.c14 tr td:first-child {
  border-left: 0.5px solid #D5D5E7;
}

.c14 tr th {
  border-top: solid 0.5px #D5D5E7;
}

.c14 tr:first-child th:last-child {
  border-top-right-radius: 16px;
}

.c14 tr:first-child th:first-child {
  border-top-left-radius: 16px;
}

.c14 tr:last-child td:first-child {
  border-bottom-left-radius: 16px;
}

.c14 tr:last-child td:last-child {
  border-bottom-right-radius: 16px;
}

.c22 tr:hover {
  background: #FAFAFF;
}

.c23 {
  min-height: 56px;
  color: #242433;
}

.c15 {
  color: #242433;
}

.c0 {
  display: block;
  width: 100%;
  border-width: 1px;
  border-style: solid;
  border-color: #D5D5E7;
  border-radius: 16px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:1366px) {
  .c1 {
    grid-gap: 24px;
  }
}

@media (min-width:768px) {

}

@media (min-width:1024px) {

}

@media (min-width:1366px) {

}

@media (min-width:1920px) {

}

@media (min-width:768px) {

}

@media (min-width:1024px) {

}

@media (min-width:1366px) {

}

@media (min-width:1920px) {

}

@media (min-width:768px) {

}

@media (min-width:1024px) {

}

@media (min-width:1366px) {

}

@media (min-width:1920px) {

}

@media (min-width:768px) {

}

@media (min-width:1024px) {

}

@media (min-width:1366px) {

}

@media (min-width:1920px) {

}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <svg
          className="c4"
          color="violet"
          fill="none"
          viewBox="0 0 28 28"
          width="28px"
        >
          <path
            clipRule="evenodd"
            d="M14 7.776a6.224 6.224 0 0 0-5.133 9.745A11.626 11.626 0 0 1 14 16.334c1.84 0 3.583.427 5.133 1.187A6.224 6.224 0 0 0 14 7.776Zm5.955 11.23A7.78 7.78 0 1 0 8.045 8.993a7.78 7.78 0 0 0 11.91 10.015Zm-1.91-.275A10.077 10.077 0 0 0 14 17.89c-1.44 0-2.807.3-4.045.84A6.2 6.2 0 0 0 14 20.225a6.2 6.2 0 0 0 4.044-1.493ZM14 10.888A1.556 1.556 0 1 0 14 14a1.556 1.556 0 0 0 0-3.112Zm-3.112 1.556a3.112 3.112 0 1 1 6.224 0 3.112 3.112 0 0 1-6.224 0Z"
            fillRule="evenodd"
          />
        </svg>
        <div
          aria-level={5}
          className="c5"
          role="heading"
          size="s"
        >
          Name: 
          Steve
        </div>
      </div>
    </div>
    <div
      className="c6"
    >
      <div
        className="c3"
      >
        <svg
          className="c4"
          color="violet"
          fill="none"
          viewBox="0 0 28 28"
          width="28px"
        >
          <path
            clipRule="evenodd"
            d="M14 7.776a6.224 6.224 0 1 0 0 12.448 6.224 6.224 0 0 0 0-12.448ZM6.22 14a7.78 7.78 0 1 1 15.56 0 7.78 7.78 0 0 1-15.56 0Z"
            fillRule="evenodd"
          />
          <path
            d="M14.097 9.626c-1.839 0-2.926.914-3.185 2.213l1.282.604c.103-1.016.79-1.608 1.813-1.608 1.087 0 1.605.476 1.605 1.21 0 .887-.66 1.35-2.357 1.556l.026 1.943h1.321l.039-1.3c1.36-.14 2.447-.848 2.447-2.392 0-1.428-1.243-2.226-2.99-2.226Zm-.155 6.909c-.596 0-.984.36-.984.913s.388.926.984.926c.582 0 .958-.373.958-.926s-.376-.913-.958-.913Z"
          />
        </svg>
        <div
          aria-level={5}
          className="c5"
          role="heading"
          size="s"
        >
          Status:
        </div>
        <div
          className="c7"
          color="green"
          size="m"
        >
          <svg
            className="c8"
            color="green"
            fill="none"
            viewBox="0 0 28 28"
            width="20px"
          >
            <path
              clipRule="evenodd"
              d="M14 7.776a6.224 6.224 0 1 0 0 12.448 6.224 6.224 0 0 0 0-12.448ZM6.22 14a7.78 7.78 0 1 1 15.56 0 7.78 7.78 0 0 1-15.56 0Zm10.664-2.106a.778.778 0 0 1 0 1.1l-3.112 3.112a.778.778 0 0 1-1.1 0l-1.556-1.556a.778.778 0 0 1 1.1-1.1l1.006 1.006 2.562-2.562a.778.778 0 0 1 1.1 0Z"
              fillRule="evenodd"
            />
          </svg>
          <span
            className="c9"
            color="inherit"
            size="m"
          >
            Ok
          </span>
        </div>
      </div>
    </div>
    <div
      className="c10"
    >
      <button
        className="c11"
        onClick={[Function]}
        size="s"
        style={
          {
            "width": "",
          }
        }
      >
        <svg
          className="c12"
          color="neutral"
          fill="none"
          viewBox="0 0 28 28"
          width="20px"
        >
          <path
            clipRule="evenodd"
            d="M19.88 8.106a1.167 1.167 0 0 0-1.65 0l-.616.616 1.65 1.65.616-.616a1.167 1.167 0 0 0 0-1.65Zm-.066 3.916 1.166-1.166a2.722 2.722 0 0 0 0-3.85l-.55.55.55-.55a2.722 2.722 0 0 0-3.85 0l-1.166 1.166v.001L6.45 17.686a.778.778 0 0 0-.228.55v2.778c0 .43.348.778.778.778h2.722a.777.777 0 0 0 .55-.228l9.541-9.541.001-.001Zm-1.65-.55-1.65-1.65-8.736 8.736v1.678H9.4l8.764-8.764Z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
    <div
      className="c13"
    >
      <div
        className=""
      >
        <table
          className="c14"
        >
          <thead
            className="c15"
          >
            <tr>
              <th
                className="c16"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="c17"
                >
                  <div
                    aria-level={6}
                    className="c18"
                    color="inherit"
                    role="heading"
                    size="xs"
                  >
                    name
                  </div>
                </div>
              </th>
              <th
                className="c19"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="c17"
                >
                  <div
                    aria-level={6}
                    className="c18"
                    color="inherit"
                    role="heading"
                    size="xs"
                  >
                    faceapi
                  </div>
                </div>
              </th>
              <th
                className="c20"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="c17"
                >
                  <div
                    aria-level={6}
                    className="c18"
                    color="inherit"
                    role="heading"
                    size="xs"
                  >
                    email
                  </div>
                </div>
              </th>
              <th
                className="c20"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="c17"
                >
                  <div
                    aria-level={6}
                    className="c18"
                    color="inherit"
                    role="heading"
                    size="xs"
                  >
                    email confirmed
                  </div>
                </div>
              </th>
              <th
                className="c21"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="c17"
                >
                  <div
                    aria-level={6}
                    className="c18"
                    color="inherit"
                    role="heading"
                    size="xs"
                  >
                    resend email
                  </div>
                </div>
              </th>
              <th
                className="c19"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="c17"
                >
                  <div
                    aria-level={6}
                    className="c18"
                    color="inherit"
                    role="heading"
                    size="xs"
                  >
                    city
                  </div>
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            className="c22"
          >
            <tr
              className="c23"
            >
              <td
                className="c24"
              >
                <p
                  className="c25"
                  color="inherit"
                  size="m"
                >
                  Steve
                </p>
              </td>
              <td
                className="c24"
              >
                <p
                  className="c25"
                  color="inherit"
                  size="m"
                >
                  <div
                    className="c7"
                    color="green"
                    size="m"
                  >
                    <svg
                      className="c8"
                      color="green"
                      fill="none"
                      viewBox="0 0 28 28"
                      width="20px"
                    >
                      <path
                        clipRule="evenodd"
                        d="M14 7.776a6.224 6.224 0 1 0 0 12.448 6.224 6.224 0 0 0 0-12.448ZM6.22 14a7.78 7.78 0 1 1 15.56 0 7.78 7.78 0 0 1-15.56 0Zm10.664-2.106a.778.778 0 0 1 0 1.1l-3.112 3.112a.778.778 0 0 1-1.1 0l-1.556-1.556a.778.778 0 0 1 1.1-1.1l1.006 1.006 2.562-2.562a.778.778 0 0 1 1.1 0Z"
                        fillRule="evenodd"
                      />
                    </svg>
                    <span
                      className="c9"
                      color="inherit"
                      size="m"
                    >
                      true
                    </span>
                  </div>
                </p>
              </td>
              <td
                className="c24"
              >
                <p
                  className="c25"
                  color="inherit"
                  size="m"
                >
                  steve@fravega.com.ar
                </p>
              </td>
              <td
                className="c24"
              >
                <p
                  className="c25"
                  color="inherit"
                  size="m"
                >
                  <div
                    className="c7"
                    color="green"
                    size="m"
                  >
                    <svg
                      className="c8"
                      color="green"
                      fill="none"
                      viewBox="0 0 28 28"
                      width="20px"
                    >
                      <path
                        clipRule="evenodd"
                        d="M14 7.776a6.224 6.224 0 1 0 0 12.448 6.224 6.224 0 0 0 0-12.448ZM6.22 14a7.78 7.78 0 1 1 15.56 0 7.78 7.78 0 0 1-15.56 0Zm10.664-2.106a.778.778 0 0 1 0 1.1l-3.112 3.112a.778.778 0 0 1-1.1 0l-1.556-1.556a.778.778 0 0 1 1.1-1.1l1.006 1.006 2.562-2.562a.778.778 0 0 1 1.1 0Z"
                        fillRule="evenodd"
                      />
                    </svg>
                    <span
                      className="c9"
                      color="inherit"
                      size="m"
                    >
                      true
                    </span>
                  </div>
                </p>
              </td>
              <td
                className="c24"
              >
                <p
                  className="c25"
                  color="inherit"
                  size="m"
                />
              </td>
              <td
                className="c24"
              >
                <p
                  className="c25"
                  color="inherit"
                  size="m"
                >
                  Buenos Aires
                </p>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
